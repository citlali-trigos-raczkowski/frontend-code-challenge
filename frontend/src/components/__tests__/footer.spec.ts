import { shallowMount, createLocalVue } from "@vue/test-utils";
import Footer from "../footer.vue";
import VueRouter from "vue-router";

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("component/footer.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(Footer, {
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("includes expected values", () => {
    const wrapper = shallowMount(Footer, {
      localVue,
    });
    expect(wrapper.text()).toContain("Thanks for visiting!");
    expect(wrapper.text()).toContain("mdi-linkedin");
    expect(wrapper.text()).toContain("mdi-github");
  });
});
