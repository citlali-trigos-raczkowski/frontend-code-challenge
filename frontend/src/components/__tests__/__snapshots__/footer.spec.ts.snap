// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/footer.vue renders correctly 1`] = `
<v-footer-stub
  dark="true"
  height="auto"
  id="footer"
  tag="footer"
>
  <v-card-title-stub
    id="footer-top"
  >
    <div
      id="footer-left"
    />
     
    <div
      id="footer-center"
    >
      <strong
        id="footer-heading"
      >
        Thanks for visiting!
      </strong>
    </div>
     
    <div
      id="footer-right"
    >
      <v-btn-stub
        activeclass=""
        class="mx-4"
        dark="true"
        icon="true"
        tag="button"
        type="button"
      >
        <a
          href="https://www.linkedin.com/in/citlali-trigos-raczkowski/"
          target="_blank"
        >
          <v-icon-stub
            color="inherit"
            id="footer-icon"
            size="24px"
          >
            
            mdi-linkedin
          
          </v-icon-stub>
        </a>
      </v-btn-stub>
      <v-btn-stub
        activeclass=""
        class="mx-4"
        dark="true"
        icon="true"
        tag="button"
        type="button"
      >
        <a
          href="https://github.com/citlali-trigos-raczkowski"
          target="_blank"
        >
          <v-icon-stub
            color="inherit"
            id="footer-icon"
            size="24px"
          >
            
            mdi-github
          
          </v-icon-stub>
        </a>
      </v-btn-stub>
    </div>
  </v-card-title-stub>
   
  <v-card-text-stub
    class="py-2 text-center"
    id="footer-bottom"
  >
    
    2021 â€” Citlali Trigos-Raczkowski
  
  </v-card-text-stub>
</v-footer-stub>
`;
